dartagnan/src/main/java/com/dat3m/dartagnan/expression/BConst.java

```diff
@@ -54,8 +54,8 @@ public boolean getBoolValue(Event e, Model model, SolverContext ctx){
 	}
 
 	@Override
-	public IConst reduce() {
-		return value ? IConst.ONE : IConst.ZERO;
+	public ExprInterface reduce() {
+		return new BConst(value);
```

Comment:
Just use ```return this```.
We should avoid creating new BConst all the time. Ideally, we would simply use a static factory method that only ever returns the same two instances (TRUE and FALSE).